.github/PULL_REQUEST_TEMPLATE/feature.md

```diff
@@ -0,0 +1,27 @@
+---
+name: Feature (PR)
+about: template for a PR that implements a new feature 
+title: "Feature: [FEATURE NAME]"
+labels: 
+assignees:
+---
+
+Technology (Ansible, Helm, Terraform): TODO
+
+### Which scenario is addressed by the change or what problem is it trying to solve?
+
+TODO
+
+
+### How is the feature being used (e.g. example code)?
+
+TODO
+
+
+### Checklist:
+
+Please tick the following before handing in your PR:
+
+* [ ] I ran `ansible-playbook` and it succeeded without any failure.
+* [ ] I ran `helm install/upgrade` and it went through with every pod being either `Running` or `Completed`.
+* [ ] I ran `terraform apply` and the expected stage was reached.
```

Comment:
Maybe we should also have CI verifying ansible and terraform code.

Reply:
Indeed! But maybe this would make sense to tackle when we actually are at the point where all the logic for infra provisioning is a) located in wsd and b) actually then consumes all the environment configuration. I mean, at that point we would anyway automate the heck out of  rolling out change

Reply:
What's `wsd` ?

When I run `ansible-playbook` I get this output:

```
ansible-playbook
Usage: ansible-playbook playbook.yml

Options:
  --ask-vault-pass      ask for vault password
  -C, --check           don't make any changes; instead, try to predict some
                        of the changes that may occur
  -D, --diff            when changing (small) files and templates, show the
                        differences in those files; works great with --check
  -e EXTRA_VARS, --extra-vars=EXTRA_VARS
                        set additional variables as key=value or YAML/JSON
  --flush-cache         clear the fact cache
  --force-handlers      run handlers even if a task fails
  -f FORKS, --forks=FORKS
                        specify number of parallel processes to use
                        (default=60)
  -h, --help            show this help message and exit
  -i INVENTORY, --inventory-file=INVENTORY
                        specify inventory host path
                        (default=/etc/ansible/hosts) or comma separated host
                        list.
  -l SUBSET, --limit=SUBSET
                        further limit selected hosts to an additional pattern
  --list-hosts          outputs a list of matching hosts; does not execute
                        anything else
  --list-tags           list all available tags
  --list-tasks          list all tasks that would be executed
  -M MODULE_PATH, --module-path=MODULE_PATH
                        specify path(s) to module library (default=None)
  --new-vault-password-file=NEW_VAULT_PASSWORD_FILE
                        new vault password file for rekey
  --output=OUTPUT_FILE  output file name for encrypt or decrypt; use - for
                        stdout
  --skip-tags=SKIP_TAGS
                        only run plays and tasks whose tags do not match these
                        values
  --start-at-task=START_AT_TASK
                        start the playbook at the task matching this name
  --step                one-step-at-a-time: confirm each task before running
  --syntax-check        perform a syntax check on the playbook, but do not
                        execute it
  -t TAGS, --tags=TAGS  only run plays and tasks tagged with these values
  --vault-password-file=VAULT_PASSWORD_FILE
                        vault password file
  -v, --verbose         verbose mode (-vvv for more, -vvvv to enable
                        connection debugging)
  --version             show program's version number and exit

  Connection Options:
    control as whom and how to connect to hosts

    -k, --ask-pass      ask for connection password
    --private-key=PRIVATE_KEY_FILE, --key-file=PRIVATE_KEY_FILE
                        use this file to authenticate the connection
    -u REMOTE_USER, --user=REMOTE_USER
                        connect as this user (default=None)
    -c CONNECTION, --connection=CONNECTION
                        connection type to use (default=smart)
    -T TIMEOUT, --timeout=TIMEOUT
                        override the connection timeout in seconds
                        (default=60)
    --ssh-common-args=SSH_COMMON_ARGS
                        specify common arguments to pass to sftp/scp/ssh (e.g.
                        ProxyCommand)
    --sftp-extra-args=SFTP_EXTRA_ARGS
                        specify extra arguments to pass to sftp only (e.g. -f,
                        -l)
    --scp-extra-args=SCP_EXTRA_ARGS
                        specify extra arguments to pass to scp only (e.g. -l)
    --ssh-extra-args=SSH_EXTRA_ARGS
                        specify extra arguments to pass to ssh only (e.g. -R)

  Privilege Escalation Options:
    control how and which user you become as on target hosts

    -s, --sudo          run operations with sudo (nopasswd) (deprecated, use
                        become)
    -U SUDO_USER, --sudo-user=SUDO_USER
                        desired sudo user (default=root) (deprecated, use
                        become)
    -S, --su            run operations with su (deprecated, use become)
    -R SU_USER, --su-user=SU_USER
                        run operations with su as this user (default=root)
                        (deprecated, use become)
    -b, --become        run operations with become (does not imply password
                        prompting)
    --become-method=BECOME_METHOD
                        privilege escalation method to use (default=sudo),
                        valid choices: [ sudo | su | pbrun | pfexec | doas |
                        dzdo | ksu ]
    --become-user=BECOME_USER
                        run operations as this user (default=root)
    --ask-sudo-pass     ask for sudo password (deprecated, use become)
    --ask-su-pass       ask for su password (deprecated, use become)
    -K, --ask-become-pass
                        ask for privilege escalation password
ERROR! You must specify a playbook file to run
```

similar to `terraform apply` - we have so far no instructions about terraform in this repo or the documentation; the terraform modules are not `terraform apply`-able as-such, but need to be imported elsewhere and applied there to test them out. 

That's not exactly useful as a checklist. Changes to helm charts never require running ansible playbooks. The PRs we got so far touched on helm charts only. We have a CI pipeline that checks 70% of the helm charts if a contributor makes a change to them. It doesn't run on external contributions as that poses a security risk. Still; the way forward here is to test more helm charts, and to test them with different versions of kubernetes; that requires a CI test matrix. A checkbox here is rarely useful, because even if someone ticks it "works for me in my environment", this doesn't tell us much.

I would remove the whole checklist section.

How about instead a line like `<!-- Have you tried out this change? Please detail the versions of helm/kubectl (if applicable) you used, and whether you can confirm it was successful (all pods "Running" or "Completed") -->`